scipy/ndimage/src/nd_image.c

```diff
@@ -334,9 +334,9 @@ static PyObject *Py_GenericFilter1D(PyObject *obj, PyObject *args)
                                         "extra_keywords must be a dictionary");
         goto exit;
     }
-    if (NpyCapsule_Check(fnc)) {
```

Comment:
This breaks backward compatibility.
On Python 2 I would support *both* pycapsule and pycobject, since it's easy to do.